engine/src/CalciteExpressionParsing.cpp

```diff
@@ -134,9 +134,19 @@ std::unique_ptr<cudf::scalar> get_scalar_from_string(const std::string & scalar_
 	if(type.id() == cudf::type_id::TIMESTAMP_DAYS) {
 		return strings::str_to_timestamp_scalar(scalar_string, type, "%Y-%m-%d");
 	}
-	if(type.id() == cudf::type_id::TIMESTAMP_SECONDS || type.id() == cudf::type_id::TIMESTAMP_MILLISECONDS
-		|| type.id() == cudf::type_id::TIMESTAMP_MICROSECONDS || type.id() == cudf::type_id::TIMESTAMP_NANOSECONDS) {
-		if (scalar_string.find(":") != std::string::npos){
+	if(type.id() == cudf::type_id::TIMESTAMP_SECONDS) {
+		if (scalar_string.find("-") != std::string::npos && scalar_string.find(":") != std::string::npos){
```

Comment:
I see here that you are checking for `-` and `:`, but it might be better to use regex. This way we know for a fact that it will match the format described